bigquery/google/cloud/bigquery/dataset.py

```diff
@@ -26,6 +27,14 @@
 from google.cloud.bigquery.table import TableReference
 
 
+_PROJECT_PREFIX_PATTERN = re.compile(
+    r"""
+    (?P<prefix>\S+\:\S+)\.+(?P<remaining>\S*)
```

Comment:
Now that we're matching this way, prefix isn't the right term. Should be `project_id`. Likewise, `remaining` should be renamed to `dataset_id`.

Also, instead of `\S`, we should be matching for characters other than `.`, that is `[^.]+`.

We want to match the whole string, so we should probably end this pattern with `$`.

Reply:
I renamed parts of the pattern, but the second comment about `[^.]` seems inappropriate to me. As we know the string could be `google.com:project.dataset`, that means that dot could be a part of the prefix. I checked couple of variants and as I see `\S` fits more.